### PR TITLE
Ported weather plugin from lion-legacy

### PR DESCRIPTION
**Pull request by Nayoshi12**
resolving issue #7. Planning to put a caching system to minimize GET requests to a few minutes.

### CODE REVIEW

File: src/app/plugins/weather.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    const out: RegExpExecArray| null = inputRegex.exec(message.toString());
+    let city: string = "";
+    if(out == null){
+        city = this.default_location;
+        key = "q";
+    }else{
+        city = out[1];
+        if(Number(out[1])){
+            key = "zip";
+        }else{
```
Comment (c650) [inline src/app/plugins/weather.plugin.ts]: Please run a code formatter (cc: @joey-colon )

File: src/app/plugins/weather.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            output += `   -   ${this.getWeatherEmoji(weather_id)} ${this.capitalize(tdata.weather[0].description)}\n`
+            xc += 1;
+        }
+        x += 1
+       
+    }
+    
+    return output
+  }
+  private getWeatherEmoji(code:number):string{
```
Comment (c650) [inline src/app/plugins/weather.plugin.ts]: Clean up all the `if`/`else` stuff by using some arrays, loops, maps or something.

**Comment by c650** [comment]
If this doesn’t get resolved soon I’m gonna take it over

**Comment by Nayoshi12** [comment]
I ran ESLint on the project and fixed all the merge issues. I am currently not able to continue with this development due to time constraints. @c650 can probably take over.

**Comment by c650** [comment]
Lgtm

**Pull request merged**